lib/compat/wordpress-5.9/theme.php

```diff
@@ -16,3 +16,13 @@ function wp_is_block_theme() {
 			is_readable( get_theme_file_path( '/templates/index.html' ) );
 	}
 }
+
+/**
+ * Note: We have to maintain this function for backward compatibility with WP 5.8.
+ * Only remove once 5.9 is the minimum supported WordPress version for the Gutenberg plugin.
```

Comment:
We could remove this line for `5.9`, as it's already suggested by structure we are trying to follow for compat changes per wp version.

I think your PR description (with a small rewording/changes) might be a better fit as a comment:

>The validate_theme_requirements method is using gutenberg_is_fse_theme in older versions of WP.